src/widgets/lv_img.c

```diff
@@ -150,6 +155,8 @@ void lv_img_set_src(lv_obj_t * obj, const void * src)
     img->cf       = header.cf;
     img->pivot.x = header.w / 2;
     img->pivot.y = header.h / 2;
+    img->dec_ctx = dec_ctx;
+    img->vector  = header.v;
```

Comment:
Why do we need `header.v`?  `dec_ctx` caps tell much more.

Reply:
By default img->dec_ctx is NULL and is only populated either by the user (via `set_src_ex`) or via the decoder (after calling `decoder_open` callback).

So until this happened, you don't know if a img is a vector image (even after you've called decoder->get_info)

Let's continue in #2926 for discussing about how to optimize this further. Ideally, decoder_get_info should allocate context and we'd be done. But that means many modifications so let's discuss what to do before going in the wrong direction.